### PR TITLE
Neural ODE is coming

### PR DESCRIPTION
**Pull request by FilippNikitin**
We offered a suitable realization of the Adjoint Method with PyTorch. The module allows implementing continuous neural networks to any PyTorch modes quickly and efficiently.    

- PyTorch autograd ODE solver 
- Adjoint Autograd 
- Continuous classifier for MNIST
- ResNet classifier for MNIST
- Example with a simple dynamic system 

To be here:
- Final notebooks
- Continuous Flow example
- Documentation

### CODE REVIEW

File: src/DeepGenerativeModels/NeuralODE/adjoint_ode.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+def zip_map(zipped, update_op):
+    return [update_op(*elems) for elems in zipped]
+
+
+def euler_update(h_list, dh_list, dt):
+    return zip_map(zip(h_list, dh_list), lambda h, dh: h + dt * dh)
+
+
+def euler_step(func, dt, state):
+    return euler_update(state, func(state), dt)
```
Comment (r-isachenko) [inline src/DeepGenerativeModels/NeuralODE/adjoint_ode.py:15]: дублируются в ноутбуке

Comment (r-isachenko) [reply src/DeepGenerativeModels/NeuralODE/adjoint_ode.py:15]: в ноутбуке нельзя оставлять комментарии, поэтому здесь напишу
- пусть plot_trajectory принимает dict: {label: trajectory}
- вынести бы переменные типа число шагов, шаг обучения и тд хотя в отдельную ячейку, а то совсем хардкод
- не хватает комментариев, очень тяжело читать (всё на англ пиши)
- не хватает секции с выводом

File: src/DeepGenerativeModels/NeuralODE/adjoint_ode.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+
+class AdjointODE(nn.Module):
```
Comment (r-isachenko) [inline src/DeepGenerativeModels/NeuralODE/adjoint_ode.py:18]: задокумментируй всё, пожалуйста, по максимуму, иначе я умру разбираться в этом всём

**Review by r-isachenko — CHANGES_REQUESTED** [review]

File: src/DeepGenerativeModels/NeuralODE/adjoint_ode.py (hunk 1 above)
Comment (newfteddy) [reply src/DeepGenerativeModels/NeuralODE/adjoint_ode.py:15]: Все комменты учтены, внес правки в последнем коммите

**Review by r-isachenko — APPROVED** [review]
